types/module.go

```diff
@@ -0,0 +1,215 @@
+package types
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/spf13/cobra"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// ModuleClients helps modules provide a standard interface for exporting client functionality
+type ModuleClients interface {
+	GetQueryCmd() *cobra.Command
+	GetTxCmd() *cobra.Command
+}
+
+// AppModule is the standard form for an application module
+type AppModule interface {
+
+	// app name
+	Name()
+
+	// registers
+	RegisterCodec(*codec.Codec)
+	RegisterInvariants(InvariantRouter)
+
+	// routes
+	Route()
+	NewHandler()
+	QuerierRoute()
```

Comment:
As a side note, can we make queriers more like the handler? Cause right now there is that whole `path[0]` thing...

Reply:
I don't know what you mean... handlers and queriers follow a very similar pattern... open up a new issue? probably should be tackled outside of this PR